docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+
+```ts
+// ...
+export async function signalClaimWork(
+  channelName: string,
+  workId: string,
+  claimantId: string,
+  userId: string,
+  say: SayFn
+) {
+  try {
+    const handle = temporalClient.workflow.getHandle(channelName);
+    await handle.signal(claimWorkSignal, {workId, claimantId});
+    console.log(`Work item ${workId} claimed by ${claimantId}`);
+    await reply(
+      say,
+      `<@${userId}> Work item ${workId} claimed by <@${claimantId}>.`
+    );
+  } catch (error) {
+    console.error("Failed to signal claim work:", error);
+  }
+}
+```
+
+<!--SNIPEND-->
+
+### Mark work item complete
+
+In the `slack_bot.ts` file, you will need to add a listener for the`wq_claim` button click event.
+
+<!--SNIPSTART typescript-slack-app-workqueue-slackbot-bot {"selectedLines": ["10", "13-14", "48-63"]}-->
+
+[docs/tutorials/typescript/work-queue-slack-app/code/bot/slack_bot.ts](https://github.com/temporalio/temporal-learning/blob/edu-2636-workqueue-slackbot-tut/docs/tutorials/typescript/work-queue-slack-app/code/bot/slack_bot.ts)
+
+```ts
+// ...
+import {
+  // ...
+  signalCompleteWork,
+} from "./modules/workqueue";
+// ...
+// Listen for Work Item Completion
+app.action<BlockAction<BlockElementAction>>(
+  "wq_complete",
+  async ({ack, say, body}) => {
+    await ack();
+    const action = body.actions[0] as ButtonAction;
+    if (action.value) {
+      const [channelName, workId, userId] = action.value.split("_");
+      const message = body.message as GenericMessageEvent;
+      // Send signal to the Temporal workflow to complete the work
+      await signalCompleteWork(channelName, workId, message, userId, say);
+    } else {
+      console.error("Action value is undefined.");
+    }
+  }
+);
+```
+
+<!--SNIPEND-->
+
+Then, in the `workqueue.ts` module, define a function `completeWork` that sends a Signal to the Work Queue Workflow to mark the work item as complete.
+
+<!--SNIPSTART typescript-slack-app-bot-workqueue-module {"selectedLines":["216-231"]}-->
+
+[docs/tutorials/typescript/work-queue-slack-app/code/bot/modules/workqueue.ts](https://github.com/temporalio/temporal-learning/blob/edu-2636-workqueue-slackbot-tut/docs/tutorials/typescript/work-queue-slack-app/code/bot/modules/workqueue.ts)
+
+```ts
+// ...
+export async function signalCompleteWork(
+  channelId: string,
+  workId: string,
+  message: GenericMessageEvent,
+  userId: string,
+  say: SayFn
+) {
+  try {
+    const handle = temporalClient.workflow.getHandle(channelId);
+    await handle.signal(completeWorkSignal, {workId});
+    console.log(`Work item ${workId} marked as complete`);
+    await reply(say, `<@${userId}> Work item ${workId} marked as complete.`);
+  } catch (error) {
+    console.error("Failed to signal complete work:", error);
+  }
+}
+```
+
+<!--SNIPEND-->
+
```

Comment:
A big missing piece here is actually running this here and showing some kind of sample output -- feels like we skipped the payoff from going right to "here's how we write all these complicated code steps" to a Conclusion.

Reply:
D'oh.
Ya totally!